utils/run_test.sh

```diff
@@ -245,12 +276,15 @@ if [ -d "/mnt/daos" ]; then
     # Reporting
     if [ "$failed" -eq 0 ]; then
         # spit out the magic string that the post build script looks for
-        echo "SUCCESS! NO TEST FAILURES"
+        produce_output 0 "SUCCESS! NO TEST FAILURES"
     else
-        echo "FAILURE: $failed tests failed (listed below)"
+        fail_msg="FAILURE: $failed tests failed (listed below)
+"
         for ((i = 0; i < ${#failures[@]}; i++)); do
-            echo "    ${failures[$i]}"
+            fail_msg=$"$fail_msg    ${failures[$i]}
+"
         done
+        produce_output 1 "$fail_msg"
```

Comment:
Just a thought, but should we only produce a single failure here even if `${#failures[@]}` > 1 or should there be a junit failure per `${failures[@]}`?

Reply:
Yeah, in most cases, we already produce junit files for things in the array, this is just to prevent us from reporting pass if anything failed.   We can look at the log to figure out what actually failed from the output here.  So I think it's best to keep it as one "test".

Reply:
> We can look at the log to figure out what actually failed from the output here

Do you mean artifacted log files or the logs that are in the Jenkins stage steps?  If the latter, I believe we are trying to move in a direction where users/developers never having to need to crack those open.  One of the ways of getting there is exposing errors/failures as JUnit results.  The functional testing framework is there.  Provisoining is not yet though.

Just want to make sure we are not taking another step backwards in that respect.

> in most cases, we already produce junit files

By most cases, are you saying we have known cases where we have tests that do not produce JUnit results?

Reply:
I think the go unit tests do 

> > We can look at the log to figure out what actually failed from the output here
> 
> Do you mean artifacted log files or the logs that are in the Jenkins stage steps? If the latter, I believe we are trying to move in a direction where users/developers never having to need to crack those open. One of the ways of getting there is exposing errors/failures as JUnit results. The functional testing framework is there. Provisoining is not yet though.
> 
> Just want to make sure we are not taking another step backwards in that respect.

I mean the log that is captured in the Jenkins stage.  We don't presently capture this log anywhere in the artifacts from what I can tell.  We save the daos logs the unit tests produce but not the stdout.

In this case, there is no step backward.  If we were to save the stdout somewhere in the artifacts, they could look at that but since we don't, this at least tells them what to look for in the unit test log.

They would get this
![image](https://user-images.githubusercontent.com/10464486/205996845-36a5eea0-a433-4f06-9a14-cb550b48081e.png)

Assuming the test failure isn't captured in another test, then they would need to look into the log but at least they know to search for vea_ut rather than first having to find the failures in the log or, worse, having the test simply pass because no junit results were produced.

> 
> > in most cases, we already produce junit files
> 
> By most cases, are you saying we have known cases where we have tests that do not produce JUnit results?

I think the go unit tests don't produce them though maybe that is a similar circumstance to this only.  I have a ticket filed on that.   But the case this is trying to fix is a known case.  The cmocka unit tests produce junit files but in this case we had a failure where the test hit a segv in initialization code so never got to the point where it would produce the junit files.   This change would at least tell us the test failed.

Reply:
> I mean the log that is captured in the Jenkins stage.

Yeah.  Those are the logs we are trying to move away from devs having to try and find.

> In this case, there is no step backward.

Well semantically, maybe not actually backwards, but it will prevent us from moving forward in a direction of not requiring devs to look at Jenkins stage/steps.

> If we were to save the stdout somewhere in the artifacts, they could look at that but since we don't, this at least tells them what to look for in the unit test log.

Understood.  But then they have to be able to find that log and their frustration level amps up when they cannot.

> 
> They would get this ![image](https://user-images.githubusercontent.com/10464486/205996845-36a5eea0-a433-4f06-9a14-cb550b48081e.png)

Again, understood, but then they have to go find the step with the log i it.

> I think the go unit tests don't produce them though maybe that is a similar circumstance to this only. I have a ticket filed on that. But the case this is trying to fix is a known case. The cmocka unit tests produce junit files but in this case we had a failure where the test hit a segv in initialization code so never got to the point where it would produce the junit files. This change would at least tell us the test failed.

So no disagreement that resolving a silent failure is great.  And I suppose this even alerts us to tests that may not be producing JUnit results.

Maybe resolving that this can result in devs having to look at stage steps will have to be a future task.